snapcraft/libraries.py

```diff
@@ -48,3 +56,40 @@ def _extract_ld_library_paths(ld_conf_file):
                 paths.extend(path_delimiters.split(line))
 
     return paths
+
+
+_libraries = None
+
+
+def _get_system_libs():
+    global _libraries
+    if _libraries:
+        return _libraries
+
+    release = platform.linux_distribution()[1]
+    lib_path = os.path.join(common.get_librariesdir(), release)
+
+    if not os.path.exists(lib_path):
+        logger.warning('No libraries to exclude from this release')
+
+    with open(lib_path) as fn:
+        _libraries = fn.read().split()
+
+    return _libraries
+
+
+def get_dependencies(elf):
+    """Return a list of libraries that are needed to satisfy elf's runtime."""
+    logger.debug('Getting dependencies for {!r}'.format(elf))
+    ldd_out = common.run_output(['ldd', elf]).split('\n')
+    ldd_out = [l.split() for l in ldd_out]
+    ldd_out = [l[2] for l in ldd_out if len(l) > 2 and os.path.exists(l[2])]
+
+    # Anything that is part of the project we don't need to check for
+    libs = list(filter(lambda x: not x.startswith(os.getcwd()), ldd_out))
```

Comment:
two (silly) things that bother me about this line: you're calling `os.getcwd` in a loop without expecting it to change, and you switched from list comprehensions to filter+lambda.